devito/ir/cgen/printer.py

```diff
@@ -149,42 +220,57 @@ def _print_SafeInv(self, expr):
 
     def _print_Mod(self, expr):
         """Print a Mod as a C-like %-based operation."""
-        args = ['(%s)' % self._print(a) for a in expr.args]
+        args = [f'({self._print(a)})' for a in expr.args]
         return '%'.join(args)
 
     def _print_Mul(self, expr):
-        term = super()._print_Mul(expr)
-        return term.replace("(-1)*", "-")
+        args = [a for a in expr.args if a != -1]
```

Comment:
I'm still kinda surprised all this tinkering is needed... why is sympy complicating our life so much?

anyway, defo an improvement from main